pkg/lib/authn/otp/kind_oob_otp.go

```diff
@@ -48,9 +48,9 @@ func (k kindOOBOTP) Purpose() Purpose {
 
 func (k kindOOBOTP) ValidPeriod() time.Duration {
 	return selectByChannel(k.channel,
-		k.config.Authenticator.OOB.Email.CodeValidPeriod,
-		k.config.Authenticator.OOB.SMS.CodeValidPeriod,
-		k.config.Authenticator.OOB.SMS.CodeValidPeriod,
+		k.config.Authenticator.OOB.Email.Deprecated_CodeValidPeriod,
+		k.config.Authenticator.OOB.SMS.Deprecated_CodeValidPeriod,
+		k.config.Authenticator.OOB.SMS.Deprecated_CodeValidPeriod,
```

Comment:
This purpose of this PR is to deprecate the fields. By deprecating, we mean we no longer use them in our code.

Now that we separate the valid period of code and the valid period of link, `kindOOBOTP` does not know whether it should use the valid period of code, or the valid period of link. So we need to add OTPForm to kindOOBOTP. It means the function `KindOOBOTP` should now take OTPForm as the 3rd argument.

With this change, you introduce many compilation errors. You visit each compilation, and reason what 3rd argument you should provide.